src/dbus_api.rs

```diff
@@ -286,7 +284,60 @@ fn destroy_filesystems(m: &MethodInfo<MTFn<TData>, TData>) -> MethodResult {
 }
 
 fn list_filesystems(m: &MethodInfo<MTFn<TData>, TData>) -> MethodResult {
-    Ok(vec![m.msg.method_return().append3("/dbus/cache/path", 0, "Ok")])
+    let message: &Message = m.msg;
+    let mut iter = message.iter_init();
+
+    if iter.arg_type() != 0 {
```

Comment:
Oh, I get it. Well that's an addition to the standard approach. What we're really complaining about is extra args...in that case we should complain about it everywhere we've filled in a method. Could you toss in a few more in the completed methods?